modules/efk-logging-collector-fluentd-v-rsyslog.adoc

```diff
@@ -0,0 +1,26 @@
+// Module included in the following assemblies:
+//
+// * logging/efk-logging-fluentd.adoc
+
+[id="efk-logging-collector-fluentd-v-rsyslog_{context}"]
+= About Fluentd and Rsyslog
+
+You can choose between Fluentd and Rsyslog as the logging collector. 
+
+Some of the noteable differences between the two collectors are:
+
+* Daching differences.
+
+* Deleted namespace
+
+* No throttling in Rsyslog. With Fluentd, an administrator can reduce the rate at which the logs are read in by Fluentd before being processed. This feature is not currently available with Rsyslog.
+
+* No secure forward in Rsyslog. With Fluentd, you can use the Fluentd the `secure-forward` plug-in to send a copy of its logs to an external log aggregator. This feature is not currently available with Rsyslog.
+
+* remote syslog
+
+When switching between the two collectors can result in:
+
+* A gap in messages between scale down of one collector and scale up of the other when reading from the tail.
+
+* Duplicate messages between scale down of one collector and scale up of the other when reading from the head.
```

Comment:
By default, rsyslog reads from the head of the container log files, and from the start of the journal, which will lead to many duplicated messages, as well as rsyslog taking a long time to "catch up" to the most current log entries.

Not sure if we should document it here, but probably in the same place where we document the comparable fluentd parameters but the env. var. `RSYSLOG_JOURNAL_READ_FROM_TAIL` (https://github.com/openshift/cluster-logging-operator/blob/master/files/rsyslog/10-viaq-modules.conf#L7) can be set `RSYSLOG_JOURNAL_READ_FROM_TAIL=true` to make rsyslog skip older entries in the journal and start reading from the end (which will lead to record loss).  Similarly for the container log files, use `RSYSLOG_FILE_READ_FROM_TAIL=true` (https://github.com/openshift/cluster-logging-operator/blob/master/files/rsyslog/60-mmk8s.conf#L11) to skip older container log records and start reading from the end of the log files (which will lead to record loss).

Reply:
We don't have anywhere we list the env vars that can be set. We have a topic that links to https://github.com/openshift/origin-aggregated-logging/blob/master/fluentd/README.md. 
I can add a link to https://github.com/openshift/origin-aggregated-logging/blob/master/rsyslog/README.md. 

For this issue ^^ I can add a line in this doc that describes the READ_FROM_TAIL variable.
```
You can configure Rsyslog to skip older entries in the journal and start reading from the end by setting the `RSYSLOG_JOURNAL_READ_FROM_TAIL`
parameter in the Rsyslog daemonset:
+
----
oc set env ds/rsyslog RSYSLOG_JOURNAL_READ_FROM_TAIL=true
----
```
Seems like the place where you would want to know about this parameter??

Reply:
ok